torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -376,6 +377,23 @@ def require_backend(backends):
     return lambda func: func
 
 
+def require_ucc_for_nccl():
+    def do_skip():
+        def is_ucc_available():
+            return dist.distributed_c10d._get_default_group().is_ucc_available
+
+        if BACKEND == dist.Backend.NCCL and not is_ucc_available():
+            raise unittest.SkipTest("Test requires UCC to be available for the NCCL backend")
+
+    def wrapper(func):
```

Comment:
no need to do it if you use `skip_if`

Reply:
I can not use `skip_if`. In order to use `skip_if`, the condition needs to be evaluated at the test-discovery time. And at that time, the default group is not initialized yet, so I don't have `is_ucc_available` available. I have to defer the evaluation of `is_ucc_available` to the time when test runs.

Reply:
Maybe alternatively, I can change the skip condition to "`TORCH_UCC_LIBRARY_PATH` is set"